config/config_test.go

```diff
@@ -0,0 +1,235 @@
+package config
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/BurntSushi/toml"
+	"github.com/stretchr/testify/assert"
+)
+
+func testLookupEnv(key string) (string, bool) {
+	mockEnv := map[string]string{
+		"GORAY":         "some_value",
+		"THREADS_COUNT": "5",
+	}
+
+	value, exists := mockEnv[key]
+	return value, exists
+}
+
+func TestInterpolate(t *testing.T) {
+	t.Run("valid: one env variable found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}")
+
+		assert.Equal(t, "some_value", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: one env variable found within the same string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "abc ${GORAY} abc")
+
+		assert.Equal(t, "abc some_value abc", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: two env variables found within the same string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}${THREADS_COUNT}")
+
+		assert.Equal(t, "some_value5", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: two env variables found within the same string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}${THREADS_COUNT}")
+
+		assert.Equal(t, "some_value5", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: no env variables within the string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${NOT AN ENV")
+
+		assert.Equal(t, "${NOT AN ENV", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("invalid: env variable not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${SOME_NAME}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME", err.Error())
+	})
+
+	t.Run("invalid: one env variable found, one not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}${SOME_NAME}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME", err.Error())
+	})
+
+	t.Run("invalid: multiple env variables not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${SOME_NAME1} ${SOME_NAME2} ${SOME_NAME3}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME1, SOME_NAME2, SOME_NAME3", err.Error())
+	})
+}
+
+func withTestReadFile(rc *ReadConf) {
+	rc.readFile = testReadFile
+}
+
+func testReadFile(name string) ([]byte, error) {
+	return []byte(`
+		[workspace]
+		version = "1.0.193"
+		root= "."
+		policies="./policy"
+		ignore-list=[".git", ".docker"]
+		go-arch="HOME: ${GORAY}"
+		
+		[analysis]
+		KRAIL1001 = {level = "info", target= "./${GORAY}"}
+		
+		[analysis.SE]
+		level = "critical"
+		target = "./internal"
+	`), nil
+}
```

Comment:
keep a test date separetly from the test code 

u can do it as a const in the top os the file